ExperimentRunner/Plugins/Garbagecollection.py

```diff
@@ -0,0 +1,73 @@
+import os.path as op
+import os
+import time
+import csv
+
+from .Profiler import Profiler
+
+
+class ConfigError(Exception):
+    pass
+
+
+class Garbagecollection(Profiler):
+    def __init__(self, config, paths):
+        super(Garbagecollection, self).__init__(config, paths)
+        self.output_dir = ''
+        self.logcat_output = ''
+        self.profile = False
+
+    def start_profiling(self, device, **kwargs):
+        self.profile = True
+        self.logcat_output = '{}logcat_{}_{}.txt'.format(self.output_dir, device.id, time.strftime('%Y.%m.%d_%H%M%S'))
+
+    def stop_profiling(self, device, **kwargs):
+        self.profile = False
+
+    def collect_results(self, device, path=None):
+        device.shell('logcat -f /mnt/sdcard/logcat.txt -d')
+        device.pull('/mnt/sdcard/logcat.txt', self.logcat_output)
+        device.shell('rm -f /mnt/sdcard/logcat.txt')
```

Comment:
@Maikerupitazu Can you check if the pull is successful before removing it?
In case it is not successful you can just raise a warning.

Reply:
Noticed that the actual crash happens later when trying to access the non-existing file when pulling fails. So instead, I check the result of the pull on whether it contains an error indication and raise a critical error log that notifies the user of missing GC measures.